crates/bevy_diagnostic/src/diagnostic.rs

```diff
@@ -45,9 +45,7 @@ impl Diagnostic {
     /// Add a new value as a [`DiagnosticMeasurement`]. Its timestamp will be [`Instant::now`].
     pub fn add_measurement(&mut self, value: f64) {
         let time = Instant::now();
-        if self.max_history_length > 1 {
-            if self.history.len() == self.max_history_length {
-                if let Some(removed_diagnostic) = self.history.pop_front() {
+        if self.max_history_length > 1 && self.history.len() == self.max_history_length && let Some(removed_diagnostic) = self.history.pop_front() {
```

Comment:
This code does not build. It also changes the logic by not performing the `self.sum -= ...` below under the same condition.

As a side note, I personally find it a lot _less_ readable as it crams 3 conditions on a single long line.

Reply:
Pardon me,my bad